### ISSUE TITLE
Spike: Billing Reports Data Analysis

### ISSUE THREAD

**Issue by mjones-oddball** [department-of-veterans-affairs/notification-api#1212]
## User Story - Business Goal
_**Required.** Please note you can have more than one story if applicable. If this is a technical investigation to achieve a larger initiative, the user story should represent that piece of the initiative._

### User Story(ies)
**As a** product manager
**I want** a high confidence in the data I use for billing
**So that** I know I will charge exactly what I should for outbound sms with amazon

**As a** business line stakeholder
**I want** to know where my notification budget is being used
**So that** I can effectively forecast and manage my team's budget

### Additional Info and Resources
- Recently, we experience discrepancies in the billing report data. Reported here: https://github.com/department-of-veterans-affairs/notification-api/issues/1141.
    - Evan discovered one source for the discrepancy, which is that we were pulling in data to BigQuery that was not billable. e.g. Twilio data does not currently have associated costs managed by our team or found in our database, failed Pinpoint messages do not incur cost and return a 0 for segments. Because we aggregate daily data to send to BQ, this creates a confusing data landscape when comparing with the audit trail in our notification API database.
    - Path forward: we are adjusting the billing integration to filter out non-billable data and want to do a clean re-assessment of data between our database and the BigQuery billing table(s) from there after it's released to production on 4/19.

## Acceptance Criteria & Checklist
_**Required**_

BVA's data showed most of the discrepancies in Evan's research. Please re-pull their sms data in production (after 4/19) and compare against their data in BigQuery table `vsp-analytics-and-insights.platform_vanotify.prod-billing`. You can pull other service data as well, but this is a good starting point to compare apples to apples.

**Given** I am comparing data between the notifications database and the bigquery billing database in production
**When** I pull billable data (cost is is greater than 0) grouped by date, service, template, billing code, and sender
**Then** I see **equal** total message counts, summed message parts, and summed costs across the 2 data sets

- [ ] Any further discrepancies found are reviewed with Product and new bug ticket(s) are created to solve them

Completed research addresses the problems and/or answers the questions below:  

- Are we 100% confident that BigQuery billing data accurately represents the outbound sms cost we're tracking in the notification database?
- Can we begin using BigQuery to generate billing reports moving forward (once a full month of clean data is present)?

## Timebox
_Leave blank if n/a_

**Comment by mjones-oddball** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/VA-Notify-620d21369d810a00146ed9c8/issues/department-of-veterans-affairs/notification-api/1212?planning-poker) with Zenhub @cris-oddball @EvanParish @jakehova @justaskdavidb2 @k-macmillan @kalbfled @ldraney

**Comment by EvanParish** [comment]
A quick update to this is that we are no longer seeing the mismatch with count and message parts, which is progress. However, I'm still seeing the occasional number that is slightly off when comparing the query we have been using directly on the db and the data we have in BigQuery.

**Comment by EvanParish** [comment]
It seems the data mismatch that was being observed is from the task getting the date and converting it to the "local" time zone, which seems to be Eastern time, then converting back to UTC when pulling from the database. This created some confusion when pulling directly from the database since the times were not aligned and therefore some of the counts were higher or lower that expected. 

Currently awaiting feedback on if a follow-up is necessary to remove the time zone conversions from the stats and billing reports being uploaded to BigQuery. A new ticket might be more effort than it's worth since the data will generally be accurate on a daily and monthly basis. Plus, as-is the data will more closely reflect what happened in Eastern time.

**Comment by EvanParish** [comment]
After speaking with @k-macmillan we won't make a follow-up ticket to change the time conversion, since as-is, it more accurately reflects what our customers would expect to see in nightly stats / billing. I'm going to add a comment about the time conversion in the [night stats doc](https://github.com/department-of-veterans-affairs/vanotify-team/blob/master/Engineering/BigQuery/Nightly_Stats_Documentation.md) and close the ticket once that is done.
